listvalidator/unique_values.go

```diff
@@ -59,10 +57,45 @@ func (v uniqueValuesValidator) ValidateList(_ context.Context, req validator.Lis
 	}
 }
 
+func (v uniqueValuesValidator) ValidateParameterList(ctx context.Context, req function.ListParameterValidatorRequest, resp *function.ListParameterValidatorResponse) {
+	if req.Value.IsNull() || req.Value.IsUnknown() {
+		return
+	}
+
+	elements := req.Value.Elements()
+
+	for indexOuter, elementOuter := range elements {
+		// Only evaluate known values for duplicates.
+		if elementOuter.IsUnknown() {
+			continue
+		}
+
+		for indexInner := indexOuter + 1; indexInner < len(elements); indexInner++ {
+			elementInner := elements[indexInner]
+
+			if elementInner.IsUnknown() {
+				continue
+			}
+
+			if !elementInner.Equal(elementOuter) {
+				continue
+			}
+
+			resp.Error = function.ConcatFuncErrors(
+				resp.Error,
+				function.NewArgumentFuncError(
+					req.ArgumentPosition,
+					fmt.Sprintf("Duplicate List Value: This attribute contains duplicate values of: %s", elementInner),
+				),
+			)
```

Comment:
nit: should we return after setting this error?

Reply:
Keeping the same implementation as the attribute validation, which actually adds multiple diagnostics.

Function errors just get concatted with a `\n`, which is ugly but matches the other implementation 😆

Reply:
Gotcha, my eyes glazed over the `ConcatFuncErrors()` function 😂